Intersect.Client/Entities/Player.cs

```diff
@@ -1618,6 +1680,58 @@ public bool TryAttack()
             return true;
         }
 
+        private void CheckAdjacentTiles()
+        {
+            AdjacentTiles.Clear(); // Clean up the list before filling it up again !
+            switch (Globals.Me.Dir)
+            {
+                case (byte)Directions.Up:
+                    AdjacentTiles.AddRange(new List<int[]>
+                    {
+                        new[] { X - 1, Y - 1 },
+                        new[] { X + 1, Y - 1 },
+                        new[] { X - 1, Y },
+                        new[] { X + 1, Y },
+                        new[] { X, Y - 1 }
+                    });
+                    break;
+                case (byte)Directions.Down:
+                    AdjacentTiles.AddRange(new List<int[]>
+                    {
+                        new[] { X - 1, Y + 1 },
+                        new[] { X + 1, Y + 1 },
+                        new[] { X - 1, Y },
+                        new[] { X + 1, Y },
+                        new[] { X, Y + 1 },
+                    });
+                    break;
+                case (byte)Directions.Left:
+                case (byte)Directions.UpLeft:
+                case (byte)Directions.DownLeft:
+                    AdjacentTiles.AddRange(new List<int[]>
+                    {
+                        new[] { X - 1, Y - 1 },
+                        new[] { X - 1, Y + 1 },
+                        new[] { X - 1, Y },
+                        new[] { X, Y - 1 },
+                        new[] { X, Y + 1 }
+                    });
+                    break;
+                case (byte)Directions.Right:
+                case (byte)Directions.UpRight:
+                case (byte)Directions.DownRight:
+                    AdjacentTiles.AddRange(new List<int[]>
+                    {
+                        new[] { X + 1, Y - 1 },
+                        new[] { X + 1, Y + 1 },
+                        new[] { X + 1, Y },
+                        new[] { X, Y - 1 },
+                        new[] { X, Y + 1 }
+                    });
+                    break;
+            }
+        }
```

Comment:
The worst part is that these directions are actually also **wrong** because for example the only adjacent tiles to UpRight are in the directions Up, UpRight, and Right. This is literally assigning 180 degrees to the "adjacency" detection for target detection. This is as though with 4 directions that if you were facing up that you could hit left, up, and right, instead of just right. The correct values would be a cone (and would require a full 8 distinct branches instead of the 4 found here), but let's pretend this abomination never existed and go for a zero-memory solution.